src/renderer/editor-mosaic.ts

```diff
@@ -150,173 +139,110 @@ export class EditorMosaic {
    * @param {EditorId} id
    */
   @action public removeCustomMosaic(id: EditorId) {
-    this.hideAndBackupMosaic(id);
-    this.editors.delete(id);
+    this.hide(id);
     this.customMosaics = this.customMosaics.filter((mosaic) => mosaic !== id);
   }
 
-  /**
-   * Shows the editor value for a given editor.
-   *
-   * @param {EditorId} id
-   */
-  @action public showMosaic(id: EditorId) {
-    this.setVisibleMosaics([...this.getVisibleMosaics(), id]);
+  /** Reset the mosaic's layout back to the default */
+  @action public resetLayout() {
+    this.set(this.values());
```

Comment:
Optional: I'm wondering if I'm overthinking this, but I had trouble understanding the magic behind the `resetLayout()` function.

`this.set` does all the heavy lifting, but you have to go a couple layers deep in the abstraction to understand how exactly the default layout is set.

Reply:
This is a good point. I'll try to make it more intuitive.

Reply:
I've tried to improve this readability in 203c6ae:

- rename addOrReplace() to addFile() for clarity
- add some comments for nonobvious parts esp. in set(), setVisible()
- inline-method refactor: move createModel() into addFile() as a minor readability tweak; one fewer function to have to follow the control-flow through when reading.
- reorganized the source to put related functions together, esp. putting resetLayout() before set() and moving show() to be right after addFile() so that one can begin reading at resetLayout(), the functions that follow in the file are the same as the call chain: resetLayout() -> set() -> addFile() -> show() -> setVisible()

Reply:
I think the comments make it a lot clearer!